### PR TITLE
Configuration for OTel support in iOS RUM

### PR DESCRIPTION
**Pull request by maciejburda**
### What does this PR do?
RUM now supports correlation with APM through Open Telemetry. This PR updates the documentation with new configuration parameters to use the functionality.

### Additional Notes
This is the PR for iOS RUM , Android will soon follow.

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Check images for PII
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

**Review by plousada — APPROVED** [review]

File: content/en/real_user_monitoring/connect_rum_and_traces/_index.md [hunk 1]
```diff
@@ -39,7 +39,7 @@ Use frontend data from RUM, as well as backend, infrastructure, and log informat
 -   Your HTTP servers are using [a library that supports distributed tracing](#supported-libraries).
 -   You have the following set up based on your SDK:
     - With the **Browser SDK**, you have added the XMLHttpRequest (XHR) or Fetch resources on the RUM Explorer to your `allowedTracingUrls`.
-    - With the **Mobile SDK**, you have added the Native or XMLHttpRequest (XHR) to your `firstPartyHosts`.
+    - With the **Mobile SDK**, you have added the Native or XMLHttpRequest (XHR) to using `trackURLSession(firstPartyHosts:)` builder function.
```
Comment (maycmlee) [inline content/en/real_user_monitoring/connect_rum_and_traces/_index.md:42]:
```suggestion
    - With the **Mobile SDK**, you have added the Native or XMLHttpRequest (XHR) to use `trackURLSession(firstPartyHosts:)` builder function.
```

File: content/en/real_user_monitoring/connect_rum_and_traces/_index.md [hunk 2]
```diff
@@ -209,7 +213,36 @@ RUM supports several propagator types to connect resources with backends that ar
       - `tracecontext`: [W3C Trace Context](https://www.w3.org/TR/trace-context/) (`traceparent`)
       - `b3`: [B3 single header](https://github.com/openzipkin/b3-propagation#single-header) (`b3`)
       - `b3multi`: [B3 multiple headers](https://github.com/openzipkin/b3-propagation#multiple-headers) (`X-B3-*`)
+{{% /tab %}}
+
+{{% tab "iOS RUM" %}}
+1. Set up RUM to connect with APM as described above.
```
Comment (maycmlee) [inline content/en/real_user_monitoring/connect_rum_and_traces/_index.md:219]:
```suggestion
1. [Set up RUM to connect with APM](#setup-rum).
```

Comment (maycmlee) [reply content/en/real_user_monitoring/connect_rum_and_traces/_index.md:219]: Is that anchor link linking to the correct section?

File: content/en/real_user_monitoring/connect_rum_and_traces/_index.md [hunk 3]
```diff
@@ -209,7 +213,36 @@ RUM supports several propagator types to connect resources with backends that ar
...
+            )
+            .trackURLSession(
+                firstPartyHostsWithHeaderTypes: [
+                    "api.example.com": [.tracecontext]
+                ]
+            )
+            .build()
+        )
+    ```
+    `trackURLSession(firstPartyHostsWithHeaderTypes:)` takes `Dictionary<String, Set<TracingHeaderType>>` as a parameter, where key is a host and value is a list of supported tracing header types.
```
Comment (maycmlee) [inline content/en/real_user_monitoring/connect_rum_and_traces/_index.md:239]:
```suggestion
    `trackURLSession(firstPartyHostsWithHeaderTypes:)` takes `Dictionary<String, Set<TracingHeaderType>>` as a parameter, where the key is a host and the value is a list of supported tracing header types.
```

File: content/en/real_user_monitoring/connect_rum_and_traces/_index.md [hunk 4]
```diff
@@ -209,7 +213,36 @@ RUM supports several propagator types to connect resources with backends that ar
...
 
+    `TracingHeaderType` in an enum representing following tracing header types:
```
Comment (maycmlee) [inline content/en/real_user_monitoring/connect_rum_and_traces/_index.md:241]:
```suggestion
    `TracingHeaderType` in an enum representing the following tracing header types:
```

**Review by maycmlee — COMMENTED** [review]
Left some feedback

**Review by plousada — CHANGES_REQUESTED** [review]

File: content/en/real_user_monitoring/connect_rum_and_traces/_index.md (hunk 2 above)
Comment (plousada) [reply content/en/real_user_monitoring/connect_rum_and_traces/_index.md:219]: If we change it here we should also change in the Browser section to make it coherent, no?

Comment (maycmlee) [reply content/en/real_user_monitoring/connect_rum_and_traces/_index.md:219]: Oh yeah, okay, let's just leave it for now.

**Review by plousada — APPROVED** [review]

**Review by maycmlee — APPROVED** [review]
LGTM! @maciejburda let me know if I can go ahead and merge.

**Comment by maciejburda** [comment]
Thank you for the review, @maycmlee. I greatly appreciate it.

I am ready for the merge. Let's move forward! 🚀

**Pull request merged**